### DISCUSSION TITLE
Make use of ssh keys instead of using passwords

### DISCUSSION THREAD

**Discussion by thnow** [josephernest/nfreezer#9] [Ideas]
I really like the idea of nfreezer, but using password is insecure and often forbidden on the servers which I use. Using ssh keys instead would be more secure and easy to handle. Would be great if it could be integrated. 

Thomas

**Comment by josephernest** [comment]
Good idea @thnow! Just for a test, can you modify line 126 (https://github.com/josephernest/nfreezer/blob/master/nfreezer.py#L126) 

    with pysftp.Connection(host, username=user, password=sftppwd) as sftp:

by

    with pysftp.Connection(host, username=user, private_key="/path/to/your/keyfile")

or

    with pysftp.Connection(host, username=user, private_key="/path/to/your/keyfile", private_key_pass="...")

just to see if it works?

**Reply by thnow** [reply]
Yes, works, directly connects.

**Comment by josephernest** [comment]
Great @thnow! Do you use it with just a key file:

    with pysftp.Connection(host, username=user, private_key="/path/to/your/keyfile")

or, with a key file + a key file password:

    with pysftp.Connection(host, username=user, private_key="/path/to/your/keyfile", private_key_pass="...")

?

**Reply by thnow** [reply]
I used it without passphrase, all my backup keys are without passphrase. (Because the passphrase would be in cleartext in the config/cronjob otherwise)

**Comment by capncrunch** [comment]
Would it be possible to modify this solution so an active SSH Agent can be used for the key?
